podzim_2025/Dragca_K/Ukoly_sprava_uzivatelu_sudo.txt

```diff
@@ -0,0 +1,278 @@
+6. Změň vlastníka souboru na uživatele eva a skupinu na studenti.
+chown eva:studenti poznamka.txt
+ale z učtu petr mi to nešlo, tak jsem se přepla na svůj účet a pak se sudo:
+sudo chown eva:studenti poznamka.txt 
+Nevím, proč mi to nešlo... že by proto, že k chown je třeba sudo a petr není ve wheel?
+7. Ověř, že soubor má nového vlastníka i skupinu.
+petr@fedora:~$ ll
+celkem 0
+-rw-r-----. 1 eva studenti 0 15. říj 21.44 poznamka.txt
+
+Úkol 3: Sdílená složka pro skupinu
+1.    Jako root vytvoř složku /home/spolecne.
+sudo mkdir /home/spolecne
+ls -l /home
+celkem 0
+drwx------. 1   1002     1004  80 15. říj 21.19 adam
+drwx------. 1 eva    studenti  80 12. říj 18.18 alena
+drwx------. 1 dragca dragca   540 15. říj 22.38 dragca
+drwx------. 1 eva    eva       80 15. říj 21.17 eva
+drwx------. 1 petr   petr     170 15. říj 22.38 petr
+drwxr-xr-x. 1 root   root       0 15. říj 23.05 spolecne
+
+2.    Nastav vlastníka root a skupinu studenti:
+sudo chgrp <studenti> </home/spolecne>
+
+drwxr-xr-x. 1 root   studenti   0 15. říj 23.05 spolecne
+
+3.    Nastav práva tak, aby: vlastník měl práva rw, skupina měla práva rw, ostatní neměli žádná.
+dragca@fedora:~$ sudo chmod u=rw,g=rw,o= /home/spolecne 
+
+drw-rw----. 1 root   studenti   0 15. říj 23.05 spolecne
+
+4.    Nastav setgid bit, aby nově vytvořené soubory ve složce zdědily skupinu studenti:
+
+    chmod g+s /home/spolecne
+    
+drw-rwS---. 1 root   studenti   0 16. říj 07.02 spolecne
+
+5.    Ve složce /home/spolecne vytvoř soubor zkouska.txt jako uživatel eva.
+ikdyž bylo nastavení, jak je vidět výše, nemohla jsem po přepnutí na uživatele eva vytvořit soubor v teto složce. Až po přidání práva execute skupině jsem pak mohla jako eva příkazem touch vytvořit soubor. Výpis složky se změnil na:
+drw-rws---. 1 root   studenti  22 21. říj 18.54 spolecne
+(z velkého S na malé)
+6.    Ověř, že soubor má skupinu studenti:
+
+-rw-r--r--. 1 eva studenti 0 21. říj 18.54 zkouska.txt
+
+7.   Přepni se na uživatele adam a zkontroluj, že může tento soubor číst i upravovat. - Adama jsem měli smazat v úkolech :praktická část 1/ukol 6
+zkusila jsem si jej znovu vytvořit, ale protože už v /home/adam složka adam byla, ikdyž uživatel byl smazaný, bash napsal: "vytvářím poštovní schránku" a pak mi nešlo přepnout se na uživatele adam, psalo mi to, že byl odepřen přístup k /home/adam.
+Proto jsem vytvořila uživatele studentka, přidala ji skupinu studenti.
+ale vzhledem k výpisu oprávnění souboru:
+
+studentka@fedora:~$ ls -l /home/spolecne/zkouska.txt
+-rw-r--r--. 1 eva studenti 0 21. říj 18.54 /home/spolecne/zkouska.txt
+
+může studenka( nebo adam) jen soubor číst, nemůže do něj zapisovat, k tomu se musím přepnout na evu (vlastníka souboru) a zadat chmod g+w zkouska.txt:
+
+-rw-rw-r--. 1 eva studenti 0 21. říj 18.54 zkouska.txt
+pak teprve můžu jako jiný uživatel ze skupiny upravovat soubor:
+studentka@fedora:~$ cat /home/spolecne/zkouska.txt
+tohle zapsala studentka
+
```

Comment:
Zadání budu muset dost předělat koukám. 
První problém, pokud se uživatel odstraňuje tak se automaticky neodstraní i jeho domovský adresář. To se dá udělat s přepínačem -r . Tzn: userdel -r adam
Pak nám tu vznikl tento problém při znovuvytváření.
Druhý problém, chybí právo zápisu skupiny do souboru.